src/components/Table/GridTable.test.tsx

```diff
@@ -2604,6 +2604,145 @@ describe("GridTable", () => {
+                  data: ({ lastName }) => lastName,
+                }),
+              ],
+            }),
+          ]}
+          rows={[
+            { kind: "header", id: "header", data: {} },
+            { kind: "expandableHeader", id: "expandableHeader", data: {} },
+            { kind: "data", id: "user:1", data: { firstName: "Brandon", lastName: "Dow", age: 36 } },
+          ]}
+          persistCollapse={tableIdentifier}
+        />,
+      );
+
+      // Then the column is initially expanded
+      expect(cell(r, 1, 0)).toHaveTextContent("First name");
+      expect(cell(r, 1, 1)).toHaveTextContent("Last name");
+    });
+
+    it("ignores init expanded, but respects new columns", async () => {
+      // Given a table with a column that is initially hidden, and initially expanded where the expanded columns are lazily loaded
+      const columns: GridColumn<ExpandableRow>[] = [
+        {
+          id: "myColumn",
+          header: emptyCell,
+          data: ({ firstName }) => firstName,
+          expandableHeader: "First Name",
+        },
+        {
+          id: "myColumn2",
+          initVisible: false,
+          initExpanded: true,
+          canHide: true,
+          header: "First",
+          data: ({ firstName }) => firstName,
+          expandableHeader: "name",
+          expandColumns: async () => [
+            column<ExpandableRow>({
+              expandableHeader: emptyCell,
+              header: "Last Name",
+              data: ({ lastName }) => lastName,
+              w: "250px",
+            }),
+            column<ExpandableRow>({
+              expandableHeader: emptyCell,
+              header: "Age",
+              data: ({ age }) => age,
+              w: "80px",
+            }),
+          ],
+        },
+      ];
+
+      const rows: GridDataRow<ExpandableRow>[] = [
+        { kind: "header", id: "header", data: {} },
+        { kind: "expandableHeader", id: "expandableHeader", data: {} },
+        { kind: "data", id: "user:1", data: { firstName: "Brandon", lastName: "Dow", age: 36 } },
+      ];
+
+      // And a table tied to the GridTableApi
+      const api: MutableRefObject<GridTableApi<ExpandableRow> | undefined> = { current: undefined };
+
+      function Test() {
+        const _api = useGridTableApi<ExpandableRow>();
+        api.current = _api;
+        return <GridTable api={_api} columns={columns} rows={rows} />;
+      }
+
+      // When rendering the table
+      const r = await render(<Test />);
+      // Then the column is initially hidden
+      expect(row(r, 1).childNodes).toHaveLength(1);
+      // When setting the column to be visible
+      api.current?.setVisibleColumns(api.current.getVisibleColumnIds().concat("myColumn2"));
+      expect(row(r, 1).childNodes).toHaveLength(2);
+      // wait for promise to resolve
+      await wait();
+
+      // then expect 2 columns + 2 expandable columns to be visible
+      expect(row(r, 1).childNodes).toHaveLength(4);
+    });
   });
```

Comment:
**praise:** Great tests! Lazy ask, do you know if we already have a test for `expandColumns` that'd be like:

```
it("ignores initExpanded for existing columns on rerender, async() => {
  // Given `columnA` that is initially expanded
  // When collapsing `columnA`
  // And when then triggering new `columnB` to be introduced.
  // Then the `columnA` remains collapsed
});
```

Reply:
I actually think this should be in a `tableState` test